### PR TITLE
feat: update CloudFormation resources v2.18.0

### PR DESCRIPTION
**Pull request by eladb**
Rewrite the create-missing-libraries script in typescript
so it's a bit more robust and mangles names nicely.

New resource types:

* AWS::AmazonMQ::ConfigurationAssociation
* AWS::IoTAnalytics::Channel
* AWS::IoTAnalytics::Dataset
* AWS::IoTAnalytics::Datastore
* AWS::IoTAnalytics::Pipeline

----

### Pull Request Checklist

Please check all boxes, including N/A items:

#### Testing

- [x] Unit test and/or integration test added
- [x] __Toolkit change?:__ [integration tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh) manually executed (paste output to the PR description)
- [x] __Init template change?:__ coordinated update of integration tests (currently maintained in a private repo).

#### Documentation

- [x] __README__: README and/or documentation topic updated
- [ ] __jsdocs__: All public APIs documented

### Title and description

- [x] __Change type__: Title is prefixed with change type:
  * `fix(module): <title>` bug fix (_patch_)
  * `feat(module): <title>` feature/capability (_minor_)
  * `chore(module): <title> ` won't appear in changelog
  * `build(module): <title>` won't appear in changelog
- [x] __Title format__: Title uses lower case and doesn't end with a period
- [x] __Breaking change?__: Last paragraph of description is: `BREAKING CHANGE: <describe exactly what changed and how to achieve similar behavior + link to documentation/gist/issue if more details are required>`
- [x] __References__: Indicate issues fixed via: `Fixes #xxx` or `Closes #xxx`

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by sam-goodwin** [comment]
Oops, didn't see that the build failed.

File: packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+ * have an AWS construct library.
+ */
+
+import child_process = require('child_process');
+import fs = require('fs');
+import path = require('path');
+import { promisify } from 'util';
+import cfnspec = require('../lib');
+
+const exist = promisify(fs.exists);
```
Comment (RomainMuller) [inline packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts]: Could have used `fs-extra` here... Would have made those unnecessary. But optional of course.

File: packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    const lowcaseModuleName = moduleBaseName.toLocaleLowerCase();
+    const packageName = `@aws-cdk/${moduleName}`;
+
+    // dotnet names
+    const dotnetPackage = `Amazon.CDK.${moduleFamily}.${moduleBaseName}`;
+
+    // java names
+    const javaGroupId = 'software.amazon.awscdk';
+    const javaPackage = moduleFamily === 'AWS'
+      ? `services.${lowcaseModuleName}`
```
Comment (RomainMuller) [inline packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts]: I like multi-line ternary assignment r-values to have the `?` and `:` column-aligned on the assignment `=`. But maybe that's just me? :)

File: packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      '```',
+    ]);
+
+    const templateDir = path.join(__dirname, 'template');
+    for (const file of await readdir(templateDir)) {
+      await copyFile(path.join(templateDir, file), path.join(pacakgePath, file));
+    }
+
+    // bootstrap and build the package and all deps to ensure integrity
+    await exec(`lerna bootstrap --scope ${packageName}`);
```
Comment (RomainMuller) [inline packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts]: Would get path to `lerna` from dependencies.

**Comment by eladb** [comment]
@sam-goodwin  It's okay to approve even if the build fails, the onus is on me to make sure it passes before merging.

File: packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts (hunk 1 above)
Comment (eladb) [reply packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts]: Done, you are right.

File: packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts (hunk 2 above)
Comment (eladb) [reply packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts]: I guess it's a matter of style. I kinda like it like this if that's okay :-)

File: packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts (hunk 3 above)
Comment (eladb) [reply packages/@aws-cdk/cfnspec/build-tools/create-missing-libraries.ts]: Done.

**Pull request merged**